### DISCUSSION TITLE
Dicas para montar servidor local

### DISCUSSION THREAD

**Discussion by didiraja** [frontendbr/forum#683] [Perguntas e Respostas]
Confrades, estou começando a me aventurar mais pelo wordpress e usando servidor local pra testar comunicação com bancos e gostaria de dicas de como fazer e, quem sabe, otimizar esse processo.

No momento, estou no Mac e uso o XAMPP, voces possuem algum recomendação melhor? Indicam alguma configuração ou programa melhor para este processo?

P.S. Sei que a pergunta é abrangente mas meu conhecimento sobre isto é baixo, então podem ser bem básicos

**Comment by tarciozemel** [comment]
Comece a usar [Docker](https://www.docker.com/).

[Aqui há um exemplo prático Docker + WP](https://www.sitepoint.com/how-to-use-the-official-docker-wordpress-image/).

**Comment by kazzkiq** [comment]
+1 pro Docker, @tarciozemel!

Eu fiquei enrolando pra começar a mexer com ele, achando que era overkill (a gente sempre arruma desculpas quando tem medo de aprender algo novo né), mas acabei me obrigando por causa do uso dele no trabalho. Hoje já não me imagino usando outra coisa.

Acabou aquele caos de ficar configurando linguagem, banco de dados e servidor na sua máquina, só pra depois descobrir que tá tudo diferente do ambiente de produção. Com Docker cada pedaço da sua aplicação roda num ambiente separado, que funciona de forma idêntica, independentemente de qual máquina ele esteja rodando.

O vídeo do @hposca é uma ótima porta de entrada pra começar a mexer com Docker, ele foi o melhor vídeo em português que eu encontrei no Youtube, e não deve absolutamente nada aos vídeos gringos:
https://www.youtube.com/watch?v=hCMcQfGb4cA

**Comment by didiraja** [comment]
Já ouvi falar muito desse docker, mas jurava que era ferramenta avançada de back-end, vou conferir esse vídeo. Brigado aos dois.

**Comment by kivervinicius** [comment]
Para php eu uso o servidor local embutido nele php -S para quando não há muita necessidade. Mas o docker é a coisa mais linda do mundo :)

**Comment by renanlara** [comment]
Eu uso o Vagrant. Tem um repositório muito legal que vem com o server bem completo: https://github.com/vagrantpress/vagrantpress

**Comment by aferreira44** [comment]
Docker e Vagrant são bem legais, mas se você quer uma solução ainda mais simples focada em Wordpress, tem o Local. Dá uma conferida aí, é bem simples mesmo e fácil de usar. https://local.getflywheel.com/

Para o Vagrant, também gosto do box da Scotch.io -> https://box.scotch.io/

**Comment by allysonsouza** [comment]
Trabalho com WordPress desde 2013, comecei com EasyPHP, e agora estou no XAMPP - no Windows. 

Trabalhei em alguns projetos razoavelmente grandes com WordPress e ele me atendeu, tentei migrar para o Vagrant mas a falta de estabilidade no início do projeto e complicação me afastaram dele. Tentei o Docker, infelizmente no meu desktop o processador não foi compátivel para virtualização na época, lembro que parece que dava para resolver, mas no fim, o bom e velho XAMPP continuava rodando até do pen drive se eu quisesse, persisto nele por hora.
